integration/cluster_test.go

```diff
@@ -371,3 +371,25 @@ func clusterMustProgress(t *testing.T, membs []*member) {
 		mcancel()
 	}
 }
+
+func TestStopTriggerTransferLeader(t *testing.T) {
```

Comment:
Wouldn't this test work without the changes in this PR? (or maybe with a `time.Sleep` before the second `waitLeader` so it catches the election?)

I think there needs to be concurrent quorum gets on a follower to correctly test leadership transfer. The gets would fail when the leader is taken down without transfer and succeed if transfer works.